### ISSUE TITLE
Support for multiple types in child list

### ISSUE THREAD

**Issue by MatthewScholefield** [dapper91/pydantic-xml#13]
It would be nice to support using `foo: List[Union[A, B]] = element()` to parse all elements into one list typed based on the tag. Here is an example from the OpenSubtitles dataset:
XML:

```xml
<?xml version="1.0" encoding="utf-8"?>
<document id="3651718">
  <s id="2">
    <time id="T2S" value="00:00:38,699" />
    <w id="2.1">in</w>
    <w id="2.2">1914</w>
    <time id="T2E" value="00:00:41,699" />
    <time id="T3S" value="00:00:43,460" />
    <w id="2.3">in</w>
    <w id="2.4">the</w>
    <w id="2.5">village</w>
    <w id="2.6">of</w>
    <w id="2.7">Ryazan</w>
    <time id="T3E" value="00:00:47,460" />
  </s>
</document>
```

Proposed Code:
```python

from typing import List, Optional, Union
from pydantic_xml import BaseXmlModel, element, attr

class TimingEntry(BaseXmlModel, tag='time'):
    id: str = attr()
    value: str = attr()

class SubtitleEntry(BaseXmlModel, tag='w'):
    id: str = attr()
    content: str

class SubtitleChunk(BaseXmlModel):
    entries: List[Union[TimingEntry, SubtitleEntry]] = element()   # tag is extracted from metaclass arguments to TimingEntry and SubtitleEntry

class Subtitles(BaseXmlModel, tag='document'):
    id: str = attr()
    chunks: List[SubtitleChunk] = element(tag='s')

Subtitles.from_xml('input.xml')
```

I tried getting this to work by passing in `element(tag=['w', 'time'])` and modifying `pydantic_xml/serializers.py:591` to match a list of tags. This almost works (`Union[Foo]` works if all entries can be deserialized to `Foo`), but for some reason putting two items in the union fails with:
```
ValidationError: 38 validation errors for Subtitles
chunks -> 0 -> entries -> 0
  value is not a valid dict (type=type_error.dict)
...
chunks -> 0 -> entries -> 10
  none is not an allowed value (type=type_error.none.not_allowed)
chunks -> 1 -> entries -> 0
  value is not a valid dict (type=type_error.dict)
...
chunks -> 1 -> entries -> 10
  none is not an allowed value (type=type_error.none.not_allowed)
```
Note that this was even with each type in the union having `content: Optional[str] = None` and `value: Optional[str] = attr()`.

Any high-level idea about how this could be implemented?

**Comment by Jacob-Flasheye** [comment]
How does Pydantic behave normally if you try to create a list of a union type?

**Comment by dapper91** [comment]
@MatthewScholefield Hi,

Due to implementation details `Union` types are not yet supported.

**Comment by dapper91** [comment]
Union type support implemented in version 0.6.0.
